tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -448,6 +452,66 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
     }
 
+    /*
+     * Self-talk test: HRR with cookie extension
+     * We also wipe the connection to ensure that the cookie stuffer is handled correctly when connections are reused.
+     */
+    {
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+
+        DEFER_CLEANUP(struct s2n_config *server_config = s2n_config_new(),
+                s2n_config_ptr_free);
+        DEFER_CLEANUP(struct s2n_config *client_config = s2n_config_new(),
```

Comment:
I think I forgot to push this comment earlier

Super minor nit: I think we could combine the two configs into one to reduce the # of lines

Reply:
The `DEFER_CLEANUP` macro is actually doing a few things and not sure its easy to combine two into one line.

The macro creates a new instance of `s2n_config` but then is also responsible for calling `s2n_config_free` at the end of the function. This is kind of equivalent to the Rust's drop method and used to free the memory associated with the object.

Reply:
Got it, thanks!